core/src/main/java/org/apache/iceberg/io/ImmutableStorageCredential.java

```diff
@@ -0,0 +1,359 @@
+   * attribute values.
+   *
+   * @return {@code true} if {@code this} is equal to {@code another} instance
+   */
+  @Override
+  public boolean equals(@Nullable Object another) {
+    if (this == another) return true;
+    return another instanceof ImmutableStorageCredential
+        && equalTo(0, (ImmutableStorageCredential) another);
+  }
+
+  private boolean equalTo(int synthetic, ImmutableStorageCredential another) {
+    return prefix.equals(another.prefix) && config.equals(another.config);
+  }
+
+  /**
+   * Computes a hash code from attributes: {@code prefix}, {@code config}.
+   *
+   * @return hashCode value
+   */
+  @Override
+  public int hashCode() {
+    @Var int h = 5381;
+    h += (h << 5) + prefix.hashCode();
+    h += (h << 5) + config.hashCode();
+    return h;
+  }
+
+  /**
+   * Prints the immutable value {@code StorageCredential} with attribute values.
+   *
+   * @return A string representation of the value
+   */
+  @Override
+  public String toString() {
+    return "StorageCredential{" + "prefix=" + prefix + ", config=" + config + "}";
+  }
+
+  private static ImmutableStorageCredential validate(ImmutableStorageCredential instance) {
+    instance.validate();
+    return instance;
+  }
+
+  /**
+   * Creates an immutable copy of a {@link StorageCredential} value. Uses accessors to get values to
+   * initialize the new immutable instance. If an instance is already immutable, it is returned as
+   * is.
+   *
+   * @param instance The instance to copy
+   * @return A copied immutable StorageCredential instance
+   */
+  public static ImmutableStorageCredential copyOf(StorageCredential instance) {
+    if (instance instanceof ImmutableStorageCredential) {
+      return (ImmutableStorageCredential) instance;
+    }
+    return ImmutableStorageCredential.builder().from(instance).build();
+  }
+
+  private Object readResolve() throws ObjectStreamException {
+    return validate(this);
+  }
+
+  /**
+   * Creates a builder for {@link ImmutableStorageCredential ImmutableStorageCredential}.
+   *
+   * <pre>
+   * ImmutableStorageCredential.builder()
+   *    .prefix(String) // required {@link StorageCredential#prefix() prefix}
+   *    .putConfig|putAllConfig(String =&gt; String) // {@link StorageCredential#config() config} mappings
+   *    .build();
+   * </pre>
+   *
+   * @return A new ImmutableStorageCredential builder
+   */
+  public static ImmutableStorageCredential.Builder builder() {
+    return new ImmutableStorageCredential.Builder();
+  }
+
+  /**
+   * Builds instances of type {@link ImmutableStorageCredential ImmutableStorageCredential}.
+   * Initialize attributes and then invoke the {@link #build()} method to create an immutable
+   * instance.
+   *
+   * <p><em>{@code Builder} is not thread-safe and generally should not be stored in a field or
+   * collection, but instead used immediately to create instances.</em>
+   */
+  @Generated(from = "StorageCredential", generator = "Immutables")
```

Comment:
Similar to above.  If we're taking over the code for this class, we should probably remove this since it's no longer generated.

Reply:
yep, let's remove this